runtime/src/tests.rs

```diff
@@ -100,3 +168,204 @@ fn pallet_poe_availability() {
         // just checking code builds, hence the pallet is available to the runtime
     });
 }
+
+mod pallets_interact {
+    use super::*;
+
+    #[test]
+    fn session_notifies_staking() {
+        new_test_ext().execute_with(|| {
+            let pre_staking_session = Staking::current_planned_session();
+            Session::rotate_session();
+            let post_staking_session = Staking::current_planned_session();
+            assert_eq!(pre_staking_session + 1, post_staking_session);
+        });
+    }
+
+    #[test]
+    fn authorship_is_configured_with_aura() {
+        new_test_ext().execute_with(|| {
+            let block_number = 1;
+            let slot = Slot::from(SLOT_ID);
+            let pre_digest = Digest {
+                logs: vec![DigestItem::PreRuntime(AURA_ENGINE_ID, slot.encode())],
+            };
+            System::reset_events();
+            System::initialize(&block_number, &Default::default(), &pre_digest);
```

Comment:
It would be nice put `authorship`'s tests in a `authorship` module and define in this module an `initialize` function that implements this logic and call it for every tests.

Reply:
Makes sense, done